okapi-core/src/main/java/org/folio/okapi/MainVerticle.java

```diff
@@ -244,93 +243,45 @@ public void start(Promise<Void> promise) {
     return tenantManager.init(vertx);
   }
 
+  @SuppressWarnings("indentation")
   private Future<Void> checkInternalModules() {
     logger.info("checkInternalModules");
-    Promise<Void> promise = Promise.promise();
     final ModuleDescriptor md = InternalModule.moduleDescriptor(okapiVersion);
     final String okapiModule = md.getId();
     final String interfaceVersion = md.getProvides()[0].getVersion();
-    moduleManager.get(okapiModule).onSuccess(gres -> {
+    return moduleManager.get(okapiModule).compose(gres -> {
       // we already have one, go on
       logger.debug("checkInternalModules: Already have {} "
           + " with interface version {}", okapiModule, interfaceVersion);
       // See Okapi-359 about version checks across the cluster
-      checkSuperTenant(okapiModule, promise);
-    }).onFailure(cause -> {
+      return Future.succeededFuture();
+    }, cause -> {
       if (OkapiError.getType(cause) != ErrorType.NOT_FOUND) {
-        promise.fail(cause); // something went badly wrong
-        return;
+        return Future.failedFuture(cause); // something went badly wrong
       }
       logger.debug("Creating the internal Okapi module {} with interface version {}",
           okapiModule, interfaceVersion);
-      moduleManager.create(md, true, true, true).onFailure(cause1 ->
-          promise.fail(cause1) // something went badly wrong
-      ).onSuccess(ires -> {
-        checkSuperTenant(okapiModule, promise);
-      });
-    });
-    return promise.future();
+      return moduleManager.create(md, true, true, true);
+    }).compose(x -> checkSuperTenant(okapiModule));
```

Comment:
This indentation allows to remove `@SuppressWarnings("indentation")`:

    return moduleManager.get(okapiModule).compose(
        gres -> {
          // we already have one, go on
          logger.debug("checkInternalModules: Already have {} "
              + " with interface version {}", okapiModule, interfaceVersion);
          // See Okapi-359 about version checks across the cluster
          return Future.succeededFuture();
        },
        cause -> {
          if (OkapiError.getType(cause) != ErrorType.NOT_FOUND) {
            return Future.failedFuture(cause); // something went badly wrong
          }
          logger.debug("Creating the internal Okapi module {} with interface version {}",
              okapiModule, interfaceVersion);
          return moduleManager.create(md, true, true, true);
        })
        .compose(x -> checkSuperTenant(okapiModule));

Reply:
Nice. Thx